src/customfilesystemmodel.cpp

```diff
@@ -169,29 +209,30 @@ bool CustomFileSystemModel::setData(const QModelIndex &index,
         }
         else if(value == Qt::Unchecked)
         {
+            // Get list of files which need manually emitting.  Must be
+            // done before modifying the _checklist.
+            QList<QModelIndex> fakeDescendents = getFakePCRecursive(index);
+
             _partialChecklist.remove(index);
             if(_checklist.remove(index))
-            {
                 emit dataChanged(index, index, selectionChangedRole);
-                if(isDir(index))
-                {
-                    // Set all children to be unchecked.
-                    for(int i = 0; i < rowCount(index); i++)
-                    {
-                        QModelIndex child = index.child(i, index.column());
-                        if(child.isValid())
-                            setData(child, Qt::Unchecked, Qt::CheckStateRole);
-                    }
-                }
-            }
+
+            // Check ancestor
             QModelIndex parent = index.parent();
-            if(parent.isValid())
+            if(parent.isValid() &&
+                    (parent.data(Qt::CheckStateRole) != Qt::Checked))
             {
                 if(hasCheckedSibling(index))
                     setIndexCheckState(parent, Qt::PartiallyChecked);
                 else
                     setIndexCheckState(parent, Qt::Unchecked);
             }
+
+            // Emit data for descendents which falsely reported PC.
+            for (int i = 0; i < fakeDescendents.count(); i++) {
```

Comment:
Why do you want to emit dataChanged for fake partially selected? Is there something to be gained apart from the test suite coverage? On the actual real code it won't make a difference and it might even force too many reloads for the View.